File: TelerikUIExamplesInCSharp/Examples/Chart/ChartTypes/FinancialChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	public class FinancialChart: ExampleViewController
	{
		TKChart chart;
		List<StockDataPoint> points;

		public override void ViewDidLoad ()
		{
			this.AddOption ("Candlestick", reloadChart);
			this.AddOption ("Ohlc", reloadChart);

			base.ViewDidLoad ();

			chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			chart.GridStyle.VerticalLinesHidden = false;
			chart.GridStyle.HorizontalLinesHidden = false;
			this.View.AddSubview (chart);

			points = StockDataPoint.LoadStockPoints (42);

			this.reloadChart (this, EventArgs.Empty);
		}

		void reloadChart(object sender, EventArgs e)
		{
			chart.RemoveAllData ();

			TKChartSeries series = null;
			if (this.SelectedOption == 0) {
				series = new TKChartCandlestickSeries (points.ToArray ());
			} else {
				series = new TKChartOhlcSeries (points.ToArray ());
			}
			series.SelectionMode = TKChartSeriesSelectionMode.DataPoint;

			TKChartNumericAxis yAxis = new TKChartNumericAxis (new NSNumber (300), new NSNumber (380));
			yAxis.MajorTickInterval = 20;
			chart.YAxis = yAxis;

			chart.AddSeries (series);

			TKChartDateTimeAxis xAxis = (TKChartDateTimeAxis)chart.XAxis;
			xAxis.MinorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Days;
			xAxis.Style.MajorTickStyle.TicksOffset = -3;
			xAxis.Style.MajorTickStyle.TicksHidden = false;
			xAxis.Style.MajorTickStyle.TicksWidth = 1.5f;
			xAxis.Style.MajorTickStyle.TicksFill = new TKSolidFill (UIColor.FromRGB (203, 203, 203));
			xAxis.Style.MajorTickStyle.MaxTickClippingMode = TKChartAxisClippingMode.Visible;
			xAxis.AllowPan = true;
			xAxis.AllowZoom = true;

			yAxis.Style.LabelStyle.TextAlignment = TKChartAxisLabelAlignment.Bottom | TKChartAxisLabelAlignment.Right;
			yAxis.AllowPan = true;
			yAxis.AllowZoom = true;

			chart.ReloadData ();
		}
	}
}

File: TelerikUIExamples/TelerikUIExamples/FinancialChart.m
//
//  CandlestickChartViewController.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//
#import <TelerikUI/TelerikUI.h>
#import "FinancialChart.h"
#import "StockDataPoint.h"

@implementation FinancialChart
{
    TKChart *financialChart;
    NSArray *_dataPoints;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self addOption:@"Candlestick" selector:@selector(reloadChart)];
        [self addOption:@"Ohlc" selector:@selector(reloadChart)];
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];

    financialChart = [[TKChart alloc] initWithFrame:CGRectInset(self.view.bounds, 10, 10)];
    financialChart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:financialChart];
    financialChart.gridStyle.verticalLinesHidden = NO;
    financialChart.gridStyle.horizontalLinesHidden = NO;
    
    _dataPoints = [StockDataPoint stockPoints:42];
    
    [self reloadChart];
}

- (void)reloadChart
{
    [financialChart removeAllData];
    
    TKChartSeries *series = nil;
    switch (self.selectedOption) {
        case 0:
            series = [[TKChartCandlestickSeries alloc] initWithItems:_dataPoints];
            break;
        case 1:
            series = [[TKChartOhlcSeries alloc] initWithItems:_dataPoints];
            break;
        default:
            break;
    }
    
    series.selectionMode = TKChartSeriesSelectionModeDataPoint;
    
    TKChartNumericAxis *yAxis = [[TKChartNumericAxis alloc] initWithMinimum:@300 andMaximum:@380];
    yAxis.majorTickInterval = @20;
    financialChart.yAxis = yAxis;
    
    [financialChart addSeries:series];
    
    TKChartDateTimeAxis *xAxis = (TKChartDateTimeAxis *)financialChart.xAxis;
    xAxis.minorTickIntervalUnit = TKChartDateTimeAxisIntervalUnitDays;
    xAxis.style.majorTickStyle.ticksOffset = -3;
    xAxis.style.majorTickStyle.ticksHidden = NO;
    xAxis.style.majorTickStyle.ticksWidth = 1.5;
    xAxis.style.majorTickStyle.ticksFill = [TKSolidFill solidFillWithColor:[UIColor colorWithRed:203/255.0 green:203/255.0 blue:203/255.0 alpha:1.0]];
    xAxis.style.majorTickStyle.maxTickClippingMode = TKChartAxisClippingModeVisible;

    financialChart.yAxis.style.labelStyle.textAlignment = TKChartAxisLabelAlignmentBottom | TKChartAxisLabelAlignmentRight;

    financialChart.xAxis.allowZoom = YES;
    financialChart.xAxis.allowPan = YES;
    financialChart.yAxis.allowZoom = YES;
    financialChart.yAxis.allowPan = YES;
}

@end
